UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Controllers/ProductDataControllerTests.cs

```diff
@@ -0,0 +1,68 @@
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using UKHO.ExchangeSetService.API.Controllers;
+using UKHO.ExchangeSetService.API.Services;
+using UKHO.ExchangeSetService.Common.Models.Request;
+using UKHO.ExchangeSetService.Common.Models.Response;
+
+namespace UKHO.ExchangeSetService.API.UnitTests.Controllers
+{
+    [TestFixture()]
+    public class ProductDataControllerTests
+    {
+        private ProductDataController controller;
+        private IHttpContextAccessor fakeHttpContextAccessor;
+        private ILogger<ProductDataController> fakeLogger;
+        private IProductDataService fakeproductDataService;
+
+        [SetUp]
+        public void Setup()
+        {
+            fakeHttpContextAccessor = A.Fake<IHttpContextAccessor>();
+            fakeproductDataService = A.Fake<IProductDataService>();
+            fakeLogger = A.Fake<ILogger<ProductDataController>>();
+
+            A.CallTo(() => fakeHttpContextAccessor.HttpContext).Returns(new DefaultHttpContext());
+            controller = new ProductDataController(fakeHttpContextAccessor, fakeLogger ,fakeproductDataService);
+        }
+
+        [Test]
+        public async Task WhenEmptySinceDateTimeInRequest_ThenGetProductDataSinceDateTimeReturnsBadRequest()
+        {
+            var validationMessage = new ValidationFailure("SinceDateTime", "Query parameter 'SinceDateTime' is required.");
+
+            validationMessage.ErrorCode = HttpStatusCode.BadRequest.ToString();
+
+            A.CallTo(() => fakeproductDataService.ValidateProductDataSinceDateTime(A<ProductDataSinceDateTimeRequest>.Ignored))
+                            .Returns(new ValidationResult(new List<ValidationFailure> { validationMessage }));
+
+            var result = (BadRequestObjectResult)await controller.GetProductDataSinceDateTime(null, "https://www.abc.com/");
+            var errors = (ErrorDescription)result.Value;
+            Assert.AreEqual(400, result.StatusCode);
+            Assert.AreEqual("Query parameter 'SinceDateTime' is required.", errors.Errors.Single().Description);
+        }
+
+        [Test]
+        public async Task WhenValidRequest_ThenGetProductDataSinceDateTimeReturnSuccess()
+        {
+
+            A.CallTo(() => fakeproductDataService.ValidateProductDataSinceDateTime(A<ProductDataSinceDateTimeRequest>.Ignored))
+                 .Returns(new ValidationResult(new List<ValidationFailure>()));
+
+            A.CallTo(() => fakeproductDataService.CreateProductDataSinceDateTime(A<ProductDataSinceDateTimeRequest>.Ignored))
+                .Returns(new ExchangeSetResponse());
+
+            var result = (OkObjectResult)await controller.GetProductDataSinceDateTime("Wed, 21 Oct 2015 07:28:00 GMT", "www.abc.com");
+            Assert.IsInstanceOf<OkObjectResult>(result);
```

Comment:
Can we do an assert that the result is the same as the response given from the CreateProductDataSinceDateTime response

Reply:
Changes has been done for result is the same as the response given from the CreateProductDataSinceDateTime response.